packages/react-router/src/ReactRouter/StackManager.tsx

```diff
@@ -59,7 +62,18 @@ export class StackManager extends React.PureComponent<StackManagerProps, StackMa
   async handlePageTransition(routeInfo: RouteInfo) {
     // If routerOutlet isn't quite ready, give it another try in a moment
     if (!this.routerOutletElement || !this.routerOutletElement.commit) {
-      setTimeout(() => this.handlePageTransition(routeInfo), 10);
+      /**
+       * The route outlet has not mounted yet. We need to wait for it to render
+       * before we can transition the page.
+       *
+       * Set a flag to indicate that we should transition the page after
+       * the component has updated.
+       */
+      this.pendingPageTransition = true;
+
+      this.setState({
```

Comment:
This may be redundant. Open to feedback for those that are more familiar with how React handles internal state changes.

Reply:
@sean-perkins - are you tossing `routeInfo` into state as a way to get another render so that `pendingPageTransition` can be evaluated?  Would it be clearer  to put `pendingPageTransition` into state (instead of an instance variable) and use it from state?

Reply:
The `componentOnReady` helper may be of use here: https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/helpers.ts#L70

Reply:
I was trying to maintain as much previous code behavior, of passing the context of `routeInfo` from the original function call, back when the function should be handled. 

Although, I don't believe `routeInfo` can change between the time React component is first constructed and the first render cycle. So I believe the `setState` is just re-assigning the same value and can be removed. I did brief testing of removing that call and tests passed and behavior seemed appropriate. 

The important piece is just catching when a page transition happens on a component that hasn't mounted (typically from routing directly to a view in a nested outlet) and re-handling that when the component is available.

Reply:
I don't believe `componentOnReady` will be applicable here. The most common state of this happening, was when `routerOutlet` (the element ref) was `undefined`. 

From testing, it does not appear to be needed in the `componentDidUpdate` life cycle 👍